### ISSUE TITLE
mindswap crashing both parties and sending the wiz back to the lobby

### ISSUE THREAD

**Issue by Kammerjunk** [d3athrow/vgstation13#26423]
#### Description of issue
wiz has mindswap (from a bundle)
wiz uses mindswap
wiz and the guy he's mindswapping both crash
the spell WORKS in that the other guy got mindswapped, but the wiz got sent back to the lobby as a new_player except he wasn't actually in the lobby system. he didn't get the lobby menu and he couldn't see the splash screen

additionally, we got a warning log about them having the same ID and CID. @ShiftyRail probably has some info about why that happens

#### Difference between expected and actual behavior
expected it to not send the wiz into lobby limbo or make me think someone's multikeying

#### Steps to reproduce
not actually sure, it doesn't always happen and I can't see any special conditions here (cause I didn't see it happen)

#### Server revision

c2562b7093b28fac6538e1b082b281e7874a4cf2

**Comment by boy2mantwicethefam** [comment]
feature

**Comment by ShiftyRail** [comment]
I *think* it had to do with the body of one of the party being destroyed/nullspaced/client disconnected.
My theory is that it went that way

ckey exchange => client A put in mob B, client B put in mob A, but mob A doesn't exist => client B is therefore deleted, booted out of the game, and can rejoin as a new player ; but a ghost has been created with the `ckey` variable being set to the `ckey` of client B, so when client B rejoins, he's sucked into this ghost.

I assume this improper exchange of clients is what causes the same ID/same CID warning, we didn't get a connection from player A with the IP/CID of player B registered on the database.

TL:DR; setting ckeys manually crashes everything

**Comment by tayachiya** [comment]
I had the same problem. As a wiz I mindswapped into HoS, crashed for sure me. But neither of our bodies were destroyed when it happend, nor did he move away

**Comment by Exxion** [comment]
I believe the fact the wizard got sent to limbo instead of the actual lobby is due to a BYOND bug where reconnecting just doesn't actually work anymore

**Comment by tayachiya** [comment]
What the hell? Why doesn't the client->reconnect not work? I know for sure it works on lifeweb, so maybe it's not entirely BYOND fault?

**Comment by Exxion** [comment]
I think it's based on server version, as it started last time we updated
If Lifeweb isn't on the same version, that would explain it
